lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -2099,60 +2139,44 @@ class CommandObjectLanguageSwiftTaskBacktrace final
-      result.AppendError("no task variable");
-      return;
-    }
-
-    StackFrame &frame = m_exe_ctx.GetFrameRef();
-    uint32_t path_options =
-        StackFrame::eExpressionPathOptionsAllowDirectIVarAccess;
-    VariableSP var_sp;
-    Status status;
-    ValueObjectSP valobj_sp = frame.GetValueForVariableExpressionPath(
-        command[0].c_str(), eDynamicDontRunTarget, path_options, var_sp,
-        status);
-    if (!valobj_sp) {
-      result.AppendError(status.AsCString());
+    llvm::Expected<ThreadSP> thread_task =
+        ThreadForTaskVariable(command, m_exe_ctx);
+    if (auto error = thread_task.takeError()) {
+      result.AppendError(toString(std::move(error)));
       return;
     }
 
-    addr_t task_ptr = LLDB_INVALID_ADDRESS;
-    ThreadSafeReflectionContext reflection_ctx;
-    if (ValueObjectSP task_obj_sp =
-            valobj_sp->GetChildMemberWithName("_task")) {
-      task_ptr = task_obj_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
-      if (task_ptr != LLDB_INVALID_ADDRESS)
-        if (auto *runtime = SwiftLanguageRuntime::Get(m_exe_ctx.GetProcessSP()))
-          reflection_ctx = runtime->GetReflectionContext();
-    }
-    if (task_ptr == LLDB_INVALID_ADDRESS || !reflection_ctx) {
-      result.AppendError("failed to access Task data from runtime");
-      return;
-    }
+    // GetStatus prints the backtrace.
+    thread_task.get()->GetStatus(result.GetOutputStream(), 0, UINT32_MAX, 0,
+                                 false, true);
+    result.SetStatus(lldb::eReturnStatusSuccessFinishResult);
+  }
+};
 
-    llvm::Expected<ReflectionContextInterface::AsyncTaskInfo> task_info =
-        reflection_ctx->asyncTaskInfo(task_ptr);
-    if (auto error = task_info.takeError()) {
-      result.AppendError(toString(std::move(error)));
-      return;
-    }
+class CommandObjectLanguageSwiftTaskSelect final : public CommandObjectParsed {
+public:
+  CommandObjectLanguageSwiftTaskSelect(CommandInterpreter &interpreter)
+      : CommandObjectParsed(
+            interpreter, "select",
+            "Change the currently selected thread to thread representation of "
+            "the given Swift Task. See `thread select`.",
+            "language swift task select <variable-name>") {
+    AddSimpleArgumentList(eArgTypeVarName);
+  }
 
-    auto thread_task = ThreadTask::Create(
-        task_info->id, task_info->resumeAsyncContext, m_exe_ctx);
+private:
+  void DoExecute(Args &command, CommandReturnObject &result) override {
+    llvm::Expected<ThreadSP> thread_task =
+        ThreadForTaskVariable(command, m_exe_ctx);
     if (auto error = thread_task.takeError()) {
       result.AppendError(toString(std::move(error)));
       return;
     }
 
-    // GetStatus prints the backtrace.
-    thread_task.get()->GetStatus(result.GetOutputStream(), 0, UINT32_MAX, 0,
-                                 false, true);
+    auto &thread_list = m_exe_ctx.GetProcessRef().GetThreadList();
```

Comment:
The Process class has a mutex to protect its thread list; should we lock it here?

Reply:
Good question. The ThreadList references the processes lock. The lock is taken inside `AddThread`, I would say there's no need to take the lock here explicitly.

Reply:
Ah, I wasn't aware that these methods were already thread-safe.
I guess the only argument for locking outside would be if these two operations need to be atomic:
```
   thread_list.AddThread(thread_task.get());
   thread_list.SetSelectedThreadByID(thread_task.get()->GetID())
```

If they don't, then no locking is needed

Reply:
I think these being non-atomic is ok. If another thread is attempting to remove the `thread_task` (which strikes me as unlikely), then I'm not sure it matters whether the removal happens before or after `SetSelectedThreadByID`.